### ISSUE TITLE
Bar colors overrides appearance colors

### ISSUE THREAD

**Issue by VimDiezel** [adriankarlen/wezterm#1]
In `wezterm.lua` if `bar.apply_to_config(c)` is the last function, it overrides the colors in `appearance.lua`, and vice versa.

**This is the default pane border color, and the cursor color is not working.**

![image](https://github.com/user-attachments/assets/72ed5bdb-5faf-4499-969a-a273100f76e2)

**This is the appearance function at the bottom, and I modified the colors in the `appearance.lua` file as shown below to override the tab bar colors.**

**And when I enable spotify to show in the bar, the whole right side disappears.**  
![{35048C6C-8C6B-4601-9B1E-8659E5575BB9}](https://github.com/user-attachments/assets/38a0894b-0cc7-498c-8e9e-bea367a14980)

![{D0957F86-F1E6-4196-ABDC-3DA40A944470}](https://github.com/user-attachments/assets/dd411fdd-5f2e-48ad-b7b0-df294a6b0b9f)

**Comment by adriankarlen** [comment]
Hi, I can recreate the issue with the colors, I am having the same issue, just did not notice it, lol.

Regarding the spotify integration with bar.wezterm, it requires you to have [spotify-tui](https://github.com/Rigellute/spotify-tui) installed.

**Comment by Lol-Lap** [comment]
> Hi, I can recreate the issue with the colors, I am having the same issue, just did not notice it, lol.
> 
> 
> 
> Regarding the spotify integration with bar.wezterm, it requires you to have [spotify-tui](https://github.com/Rigellute/spotify-tui) installed.

Well is there is a better way to fix other than what I did this hacky way. I have [Spotify-Player](https://github.com/aome510/spotify-player) Not spotify-tui unfortunately because it's so buggy and pretty much dead



Thank you for the quick response. Is there a better way to fix this than the hacky solution I used? I’m using [Spotify-Player](https://github.com/aome510/spotify-player), not spotify-tui, unfortunately, because it’s very buggy and essentially abandoned.

**Comment by VimDiezel** [comment]
> Hi, I can recreate the issue with the colors, I am having the same issue, just did not notice it, lol.
> 
> 
> 
> Regarding the spotify integration with bar.wezterm, it requires you to have [spotify-tui](https://github.com/Rigellute/spotify-tui) installed.

Thank you for the quick response. Is there a better way to fix this than the hacky solution I used? I’m using [Spotify-Player](https://github.com/aome510/spotify-player), not spotify-tui, unfortunately, because it’s very buggy and essentially abandoned.

**Comment by adriankarlen** [comment]
Yeah I am using spotify-player for tui, however what is stopping me from using `spotify-player` in bar.wezterm is this issue https://github.com/aome510/spotify-player/issues/455.

So atm I am using spotify-tui since their cli just works

**Comment by VimDiezel** [comment]
> Yeah I am using spotify-player for tui, however what is stopping me from using `spotify-player` in bar.wezterm is this issue https://github.com/aome510/spotify-player/issues/455.
> 
> 
> 
> So atm I am using spotify-tui since their cli just works

Okay, I will give spotify-tui a shot. Is it worth it? Is there a better way to fix the colors issues atm rather than the hacky solution I used?

**Comment by adriankarlen** [comment]
I don't know, there seems to be some issue with the way I am merging the configs.

The only usage I currently have of it is the spotify-integration in bar.wezterm. In all other cases i use spotify-player

**Comment by VimDiezel** [comment]
> I don't know, there seems to be some issue with the way I am merging the configs.
> 
> The only usage I currently have of it is the spotify-integration in bar.wezterm. In all other cases i use spotify-player

Okay, thank you. And please, let me know if you made any changes to fix it.
